### PR TITLE
(PEAR-1586): Update Mantine to 7

### PR DESCRIPTION
**Pull request by ocshawn**
## Description

- Update:
  - "@mantine/core": "^7.8.1",
  - "@mantine/dates": "^7.8.1",
  - "@mantine/form": "^7.8.1",
  - "@mantine/hooks": "^7.8.1",
  - "@mantine/notifications": "^7.8.1",
  - "react-plotly.js": "^2.6.0",
  - "postcss": "^8.4.38",
  - "postcss-preset-mantine": "^1.15.0",
  - "postcss-simple-vars": "^7.0.1",
- Remove StyleGuideApp unused files 

## Checklist

- [x] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [ ] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

File: packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx [hunk 1]
```diff
@@ -157,25 +157,13 @@ const ContinuousBinningModal: React.FC<ContinuousBinningModalProps> = ({
...
-  }, [intervalForm.values]);
+  }, [intervalForm.getValues()]);
 
   useEffect(() => {
     if (binMethod === "interval") {
       rangeForm.clearErrors();
-      intervalForm.validate();
     } else {
       intervalForm.clearErrors();
-      rangeForm.validate();
```
Comment (amlehman) [inline packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: This change is causing some incorrect behavior in the binning modal. If you switch modes, then you can save bins that don't follow our binning rules. Also, the last row is getting dropped? Not sure if that is due to this change or something else in the file:

https://github.com/NCI-GDC/gdc-frontend-framework/assets/4624053/771f35cb-4420-42cf-9a10-943f45ca3b36

File: packages/portal-proto/src/features/cDave/Controls.tsx [hunk 2]
```diff
@@ -151,36 +157,18 @@ const FieldControl: React.FC<FieldControlProps> = ({
...
-                  backgroundColor: theme.fn.darken(
-                    tailwindConfig.theme.extend.colors[
-                      COLOR_MAP[field.field_type]
-                    ]?.DEFAULT,
-                    0.05,
-                  ),
-                },
-              },
-              input: {
-                "&:checked + .mantine-Switch-track": {
```
Comment (amlehman) [inline packages/portal-proto/src/features/cDave/Controls.tsx:166]: These hover colors seem a bit darker than the existing ones, especially the orange. Is that an intentional style change?
Screenshot 2024-05-09 at 12 40 40 PM

**Comment by amlehman** [comment]
A few visual changes I noticed:

**Fixed** The set modal text field is smaller making the whole modal ... squatter.
Screenshot 2024-05-09 at 12 55 26 PM
Screenshot 2024-05-09 at 12 55 37 PM

**Fixed** The radio buttons on CDave aren't aligned correctly.
Screenshot 2024-05-09 at 12 25 04 PM
Screenshot 2024-05-09 at 12 24 43 PM

**Fixed** There's a right margin/padding on the title in both versions of the binning modal on CDave:
Screenshot 2024-05-09 at 1 13 25 PM
Screenshot 2024-05-09 at 1 13 33 PM
Screenshot 2024-05-09 at 1 13 39 PM
Screenshot 2024-05-09 at 1 13 46 PM

**Fixed** This collapse button on CDave has a border now.
Screenshot 2024-05-09 at 1 16 43 PM
Screenshot 2024-05-09 at 1 16 50 PM

File: packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx (hunk 1 above)
Comment (ocshawn) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: will look into, it was not intentional to remove anything

Comment (ocshawn) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: pushed a fix, the problem was the `      rangeForm.clearFieldError(
        ranges.${rangeForm.values.ranges.length - 1}.to,
      );`
no longer worked as the validation was being done asynchronously 

but it looks to work fine with just putting the checks on the top part of the form as the bottom checks every time you add a value, let me know if this works for you

File: packages/portal-proto/src/features/cDave/Controls.tsx (hunk 2 above)
Comment (ocshawn) [reply packages/portal-proto/src/features/cDave/Controls.tsx:166]: i was trying to find a suitable color that was already a preset, open to suggestions if this is too dark

**Comment by craigrbarnes** [comment]
**Fixed** Analysis Center: slight sizing issue with the Core Tools cards. The image below overlays this branch ( Mantine 7) on top of  develop. It appears the Core Tool cards are 6 pixels shorter.

image

**Comment by craigrbarnes** [comment]
Mutation Frequency Gene Table (and same with Mutation Table):

1.  **Fixed** The unselected Survival Plot checkbox fill color should not be blue
2.  **Cannot reproduce** The link icon to the left of the Gene Symbol is missing

image

**Comment by craigrbarnes** [comment]
**Fixed** Project Center more space between buttons and the top of the table. Label "TOTAL OF 83 PROJECTS" not vertically aligned with Search TextInput

image

**Comment by craigrbarnes** [comment]
Repository: 

- **Fixed** There is a slight offset with the table, and the Vertical alignment is slightly shifted down.
- **Cannot Reproduce** The link icons are missing
- 
image

**Comment by craigrbarnes** [comment]
Slide image viewer:
**Fixed** There is small gray box that's appearing at the bottom of the slide column

image

**Comment by craigrbarnes** [comment]
Gene Sumnmary: minor nitpick:

**Fixed** more/less button in Description row:  vertical alignment is shitded down vertically. 

MoreAlignment

File: packages/portal-proto/src/features/cDave/Controls.tsx (hunk 2 above)
Comment (amlehman) [reply packages/portal-proto/src/features/cDave/Controls.tsx:166]: It looks good to me 🤷‍♀️ But probably @cemile-ctds 's call.

File: packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx (hunk 1 above)
Comment (amlehman) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: Hey @ocshawn, I'm still seeing the same issue: if you switch to the range form when you have interval validation errors and then switch back to the interval form, you can save the bins when you shouldn't be able to. I'll look at the mantine form docs and see if there's a new way to replicate what these lines were doing 🤔

Comment (ocshawn) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: @amlehman thanks for looking, i think the whole form may need to be redone, or we could just remove this section of the code that removed the errors and let the errors stay in the grayed out area

Comment (amlehman) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: Hey, @ocshawn. Yeah, I'm not finding an easy way to fix the behavior. I think what we do with it is @wteouchicago 's call. There were a ton of requirements for the way this form/validation behaves, I'm not sure what is negotiable.

**Comment by wteouchicago** [comment]
Some things I noticed. 

1. **Fixed** The Send Feedback modal has more padding at the top (and also the bottom?) compared to develop. I think it's happening to other modals as well, e.g. Create...Set modal from Manage Sets (see number 2). 
- Left is develop, right is this PR.

Screenshot 2024-05-13 at 10 07 45 AM

2. **Fixed** The underline on the tab in the Create..Set modal is now a box. 
Screenshot 2024-05-13 at 10 11 50 AM

3. **Fixed** In the Repository, when the spinners on buttons show up, the text is not displayed. Guessing this will happening to all buttons with spinners
- click on Add All Files to Cart and Remove All From Cart to get the spinner to show up or click on a dropdown option in Download Associated Data

Screenshot 2024-05-13 at 10 30 00 AM

Screenshot 2024-05-13 at 10 31 37 AM

4. **Fixed** cDAVE Search on the left, the X has a box around it. I also can't seem to click on it to get rid of the search text.
Screenshot 2024-05-13 at 10 33 13 AM

File: packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx (hunk 1 above)
Comment (wteouchicago) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: Hmm, so it's one thing that we allow users to save bins that make no sense, but another thing that we seem to be showing the wrong info after allowing them to save "bad" bins (see attachment -- that case count of 75 should belong to the bin named "20-30".

\<non-technical hand-waving> can we do something like this: always revalidate all the fields when the user clicks Save Bins and don't save the bins if there are errors? Let's Slack on this if it gets complicated.

https://github.com/NCI-GDC/gdc-frontend-framework/assets/73256434/d1ea93e8-2d12-4400-945c-251c5a1bc213

Comment (ocshawn) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: i uploaded one potential fix, you can still see the error messages in the grayed out area, if this does not work i propose we hide the error messages with css when the area is not selected. @wteouchicago what do you think?

Comment (wteouchicago) [reply packages/portal-proto/src/features/cDave/ContinuousBinningModal/ContinuousBinningModal.tsx:168]: @ocshawn this works great! I think it's fine that users can still see error messages in the greyed out area. 👍

**Comment by wteouchicago** [comment]
looks good to me -- I also went over issues others reported. nice work! 👍

File: packages/portal-proto/src/features/proteinpaint/OncoMatrixWrapper.unit.test.tsx [hunk 3]
```diff
@@ -45,8 +45,8 @@ test("OncoMatrix arguments", () => {
     <MantineProvider
       theme={{
         colors: {
-          primary: ["1", "2", "3", "4", "5", "6", "7", "8", "9"],
-          base: ["1", "2", "3", "4", "5", "6", "7", "8", "9"],
+          primary: ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"],
```
Comment (craigrbarnes) [inline packages/portal-proto/src/features/proteinpaint/OncoMatrixWrapper.unit.test.tsx:48]: Good catch

**Review by craigrbarnes — APPROVED** [review]
I'm okay with the changes to support V7. Great job on implementing such a large-scale change..

**Pull request merged**